### PR TITLE
Remove Text Search Limit

### PR DESCRIPTION
**Pull request by tonysun83**

### CODE REVIEW

**Comment by rnewson** [comment]
we still need to respect the configured limit. CQ is another path to search indexes but it should honor the same configuration parameters. See https://github.com/cloudant/dreyfus/blob/master/src/dreyfus_httpd.erl#L68-69 for how dreyfus does it.

File: src/mango_cursor_text.erl [hunk 1]
```diff
@@ -49,11 +49,13 @@ create(Db, Indexes, Selector, Opts0) ->
     end,
 
     Opts = unpack_bookmark(Db#db.name, Opts0),
-
-    % Limit the result set size to 50 for Clouseau's
-    % sake. We may want to revisit this.
-    Limit0 = couch_util:get_value(limit, Opts, 50),
-    Limit = if Limit0 < 50 -> Limit0; true -> 50 end,
+    DreyfusLimit = list_to_integer(config:get("dreyfus", "max_limit", "200")),
+    Limit = case couch_util:get_value(limit, Opts, 50) of
```
Comment (rnewson) [inline src/mango_cursor_text.erl]: maybe `Limit  = erlang:min(DreyfusLimit, couch_util:get_value(limit, Opts, 50))` ?

**Comment by rnewson** [comment]
not sure why get_limit insists on lower limit of 25, so why isn't that function simply `erlang:min(get_dreyfus_limit(), CAcc#cacc.limit + CAcc#cacc.skip).` ? (cc @davisp)

**Comment by davisp** [comment]
Probably just didn't realize there was a standard config parameter for it. I'm fine with this change.

+1

**Comment by davisp** [comment]
I'm ok removing the 25 lower limit as well. Not sure what my reasoning was for that.

Also "We may want to revisit this." means I was probably aware that it was silly but didn't want to think that hard while working through the other bits.

File: src/mango_cursor_text.erl [hunk 2]
```diff
@@ -286,13 +283,18 @@ update_query_args(CAcc) ->
 
 get_limit(CAcc) ->
     Total = CAcc#cacc.limit + CAcc#cacc.skip,
```
Comment (rnewson) [inline src/mango_cursor_text.erl]: change function to `erlang:min(get_dreyfus_limit(), CAcc#cacc.limit + CAcc#cacc.skip).`

**Comment by rnewson** [comment]
+1 after that last change and squash.

**Pull request merged**